src/services/codefixes/unusedIdentifierFixes.ts

```diff
@@ -0,0 +1,153 @@
+                        if (initializer.declarations.length === 1) {
+                            return createCodeFix("", initializer.pos, initializer.end - initializer.pos);
+                        }
+                        else {
+                            if (initializer.declarations[0] === token.parent) {
+                                return createCodeFix("", token.parent.pos, token.parent.end - token.parent.pos + 1);
+                            }
+                            else {
+                                return createCodeFix("", token.parent.pos - 1, token.parent.end - token.parent.pos + 1);
+                            }
+                        }
+                    }
+                    else if (token.parent.parent.parent.kind === SyntaxKind.ForInStatement) {
+                        const forInStatement = <ForInStatement>token.parent.parent.parent;
+                        const initializer = <VariableDeclarationList>forInStatement.initializer;
+                        return createCodeFix("{}", initializer.declarations[0].pos, initializer.declarations[0].end - initializer.declarations[0].pos);
+                    }
+                    else if (token.parent.parent.parent.kind === SyntaxKind.ForOfStatement) {
+                        const forOfStatement = <ForOfStatement>token.parent.parent.parent;
+                        const initializer = <VariableDeclarationList>forOfStatement.initializer;
+                        return createCodeFix("{}", initializer.declarations[0].pos, initializer.declarations[0].end - initializer.declarations[0].pos);
+                    }
+                    else if (token.parent.parent.kind === SyntaxKind.CatchClause) {
+                        const catchClause = <CatchClause>token.parent.parent;
+                        const parameter = catchClause.variableDeclaration.getChildren()[0];
+                        return createCodeFix("", parameter.pos, parameter.end - parameter.pos);
+                    }
+                    else {
+                        const variableStatement = <VariableStatement>token.parent.parent.parent;
+                        if (variableStatement.declarationList.declarations.length === 1) {
+                            return createCodeFix("", variableStatement.pos, variableStatement.end - variableStatement.pos);
+                        }
+                        else {
+                            const declarations = variableStatement.declarationList.declarations;
+                            if (declarations[0].name === token) {
+                                return createCodeFix("", token.parent.pos + 1, token.parent.end - token.parent.pos);
+                            }
+                            else {
+                                return createCodeFix("", token.parent.pos - 1, token.parent.end - token.parent.pos + 1);
+                            }
+                        }
+                    }
+                }

```

Comment:
what about binding patterns? can you add tests for these cases. both for parameters and variable declarations.